README
npm set registry http://npm.base.wolfspool.at

( npm adduser --registry http://npm.base.wolfspool.at )

npm login (rogera/1boris)
npm publish


CHANGES
Version 0.19.6:
 - patch 0.19.6

Version 0.19.5:
 - bump pj version
 - include node modules in  vc
 - generate .gitignore
 - include version in start-log-lines
 - 0.19.4

Version 0.19.3:
 - de bundledDependencies

Version 0.19.2:
 - pj bump semver

Version 0.19.1:
 - pj bump
 - bundledDependencies array

Version 0.19.0:
 - pj bump v 0.19.0
 - bundleDependencies

Version 0.18.5:
 - bump pj vers, publishConfig to private repo

Version 0.18.4:


Version 0.18.3:
 - haendisch
 - index.css flow none, 60% col

Version 0.18.2:
 - haendisch
 - enable index.html loading

Version 0.18.1:
 - haendisch
 - via chrome / interactive

Version 0.18.0:
 - haendisch

Version 0.17.0:
 - buchhaltung
 - Merge branch 'develop' of github.com:wolfhesse/ase-whbase into develop
 - on core
 - cosmetix
 - prepare 0.17.0
 - newline stuff.. working on that CRLF/CR
 - clearfix
 - Merge tag '0.16.0' into develop
 - Merge branch 'r-0.16.0'
 - buchhaltung
 - buchhaltung

Version 0.16.0:
 - wp
 - cp phpstorm session enable..
 - update node modules..
 - so, kein unittest, teuer.. #typo
 - so, kein unittest, teuer.. #typo
 - so, kein unittest, teuer.. #typo
 - to-helo-logdump ase-whbase .. proto, logdump .. verb
 - demo latency

Version 0.15.0:
 - buchhaltung
 - 0.15.0 package
 - 0.13.2

Version 0.14.0:
 - merged master
 - srv-id to srv_id for JSON
 - 0.12.2 manual

Version 0.13.1:
 - version package info up

Version 0.13.0:
 - srv-id to srv_id for json

Version 0.12.1:
 - better dumps of json

Version 0.12.0:
 - wp

Version 0.11.0:
 - wp corr emit, comma comma comma!
 - wp

Version 0.9.0:
 - wp

Version 0.8.0:
 - emit helo --backchannel:mousemove

Version 0.7.4:
 - cleanup

Version 0.7.3:
 - wp
 - Merge branch 'develop'
 - wp styles zulu

Version 0.7.2:
 - versioning_w_style

Version 0.7.1:
 - bumpversion darf package.json nicht vergessen!..

Version 0.7.0:


Version 0.4.0:
 - version upd

Version 0.3.0:
 - package.json additions
 - touched version
 - adding index.html
 - corr name

Version 0.2.0:
 - wp

Version 0.1.0
 - intial



server.js
var app = require('http').createServer(handler)
    , io = require('socket.io').listen(app)
    , fs = require('fs');
var port = process.env.PORT || 44000;
var host = process.env.HOST || "0.0.0.0";
app.listen(port, host);

console.log("listening on "+host+":"+port);

function handler(req, res) {
    var base = '/io_original_server' // wegzunehmen
    url_minus_base = req.url.replace(base, '');
    fs.readFile(__dirname + url_minus_base, /* '/index.html', */

        function (err, data) {
//                console.log(req);
            if (err) {
                console.error(err.message);
//                    console.log(err.message);
                res.writeHead(500);
                return res.end('Error loading index.html');
            }

            res.writeHead(200, 'Content-type: text/html');
            res.end(data);
        });
}


var ev = require("events");
var em = new ev.EventEmitter();


//var chost=process.env.HOST || "0.0.0.0";
//var cport = process.env.PORT || 4000;
//var client = require("socket.io-client").connect(cport,chost);
//var client = require("socket.io-client").connect("http://0.0.0.0:4000");
// r20150227 var client = require("socket.io-client").connect("http://10.0.0.13:49153");


// versuch kerpen
var client = require("socket.io-client").connect("http://kerpen.sfe.tv:4000");

client.on("connect", function () {
    console.log("client connected");
    client.on("CONFIG", function (data) {
        console.log("client:CONFIG");
        //console.log(data);
        em.emit('/process/config', {srv: 'config', ts: new Date, data: data});
    });
    client.on("/u0/SCHEDULE", function (data) {
        console.log(data);
        em.emit('/process/schedule', {srv: 'schedule', ts: new Date, data: data});
    });
    client.on("CMD", function (data) {
        console.log("client:CMD");
        console.log(data.cmd);
        em.emit('/process/cmd', {srv: 'cmd', ts: new Date, data: data});
    });
});


io.sockets.on('connection', function (socket) {
    em.on("/process/config", function (data) {
        socket.emit("/news/global", data);
    });
    em.on("/process/schedule", function (data) {
        socket.emit("/news/business", data);
    });
    em.on("/process/cmd", function (data) {
        socket.emit("/news/business", data);
    });

    socket.on('mousemove', function (data) {
        console.log('mousemove', data);
        data.date = new Date();
        socket.volatile.emit('/news/local', data);
        setTimeout(function () {
            socket.broadcast.emit('/news/global', data);
        }, 250);
    });

    socket.on('/request/broadcast-date', function (data) {
	    console.log(data);
	    console.log(JSON.parse(data));
        console.log(data.sender);
        socket.broadcast.emit('/response/date', {date: new Date()});
    });

    socket.on('/request/date', function (data) {
        setTimeout(function () {
            socket.volatile.emit('/response/date', {date: new Date()});
        }, 2500);
    });
});



index.js
var server = module.exports.server = 'n/a';
var serve = module.exports.serve = function () {
    var app = require('http').createServer(http_handler)
        , io = require('socket.io').listen(app)
        , fs = require('fs');
    var port = process.env.PORT || 44000;
    var host = process.env.HOST || "0.0.0.0";
    app.listen(port, host);
    console.log("server listening on " + host + ":" + port);

    var version = "n/a";
    fs.readFile(__dirname + "/VERSION", /* '/index.html', */
        function (err, data) {
            if (err) {
                console.error(err.message);

            }
            version = data;
            console.log("server version " + version);
        });

    function http_handler(req, res) {
        var base = '/io_original_server'; // wegzunehmen
        url_minus_base = req.url.replace(base, '');
        if (url_minus_base === "") {
            url_minus_base = '/index.html';
        }
        if (url_minus_base === "/") {
            url_minus_base = '/index.html';
        }
        console.log(url_minus_base);
        fs.readFile(__dirname + url_minus_base, /* '/index.html', */

            function (err, data) {
                //                console.log(req);
                if (err) {
                    console.error(err.message);
                    //                    console.log(err.message);
                    res.writeHead(500);
                    return res.end('Error loading index.html');
                }

                res.writeHead(200, 'Content-type: text/html');
                res.end(data);
            });
    }


    var ev = require("events");
    var em = new ev.EventEmitter();

    console.log('server connecting to backbone ' + module.exports.server);
    var backbone = require("socket.io-client").connect(module.exports.server);
    console.log('backbone should be connected... sooon');

    var backbone_connected = false;

    em.on('to-helo-mousemove', function (data) {
        console.log("to-helo-mousemove", JSON.stringify(data));
        data['srv_id'] = 'hck-socket-d';
        if (backbone_connected)
            backbone.emit("helo", data);
    });
    em.on('to-helo-logdump', function (data) {
        console.log("to-helo-logdump", JSON.stringify(data));
        data['protocol'] = 'ase-whbase';
        data['verb'] = 'logdump';
        if (backbone_connected)
            backbone.emit("helo", data);
    });

    // listening side
    backbone.on("connect", function () {
        backbone_connected = true;
        console.log("backbone connected");
    });
    backbone.on("disconnect", function () {
        backbone_connected = false;
        console.log("backbone disconnected");
    });

    backbone.on("data", function (data) {
        console.log('backbone data ' + JSON.stringify(data));
        em.emit("/process/data", data);
    });

    backbone.on("event", function (data) {
        console.log('backbone event ' + JSON.stringify(data));
        em.emit("/process/event", data);
    });

    backbone.on("tick", function (data) {
        console.log('backbone tick ' + JSON.stringify(data));
    });


    // sending side
    //
    io.sockets.on('connection', function (socket) {
        console.log("client connected");

        em.on("/process/data", function (data) {
            socket.emit("/news/global", data);
        });

        em.on("/process/event", function (data) {
            socket.emit("/news/business", data);
        });

        socket.on('mousemove', function (data) {
            console.log('mousemove', data);
            data.date = new Date();
            socket.volatile.emit('/news/local', data);
            //setTimeout(function () {
            //    socket.broadcast.emit('/news/global', data);
            //}, 250);
            em.emit("to-helo-mousemove", data);
        });

        socket.on('/request/broadcast-date', function (data) {
            console.log(data);
            console.log(JSON.parse(data));
            console.log(data.sender);
            socket.broadcast.emit('/response/date', {date: new Date()});
        });

        socket.on('/request/date', function (data) {
            setTimeout(function () {
                socket.volatile.emit('/response/date', {date: new Date()});
            }, 2500);
        });

        socket.on('/request/logdump', function (data) {
            var logdump_latency = 100;
            setTimeout(function () {
                socket.volatile.emit('/response/logdump', {date: new Date(), purpose: 'logdump_latency demo'});
                em.emit('to-helo-logdump', {date: new Date()});
            }, logdump_latency);
        });

    });
};

